src/frontend/src/styles/main.css

```diff
@@ -794,6 +797,133 @@ a:hover,
   box-shadow: 0 0 0 4px rgba(255, 255, 255, 0.4);
 }
 
+/**
+ *  irregularity component
+ */
+.c-irregularity {
+  display: flex;
+  align-items: stretch;
+  gap: calc(var(--rs-card-bezel-x) * 0.25);
+
+  --state-color: var(--rc-warningLight);
+  --state-onColor: var(--rc-warning);
+
+  position: relative;
+  padding: calc(var(--rs-card-bezel) * 0.35) calc(var(--rs-card-bezel-x) * 0.35);
+  border: var(--vs-line) solid var(--rc-line);
+  background-color: var(--state-color);
+  border-radius: var(--vs-border-radius);
+  overflow: hidden;
+  box-shadow: 0 0 0 1px var(--state-onColor);
+}
+
+.c-irregularity + .c-irregularity {
+  margin-top: calc(var(--vs-stack) * 0.5);
+}
+
+.c-irregularity__icon {
+  position: relative;
+  flex: 0 0 4.5rem;
+  background: var(--state-onColor);
+  margin: calc(var(--rs-card-bezel) * -0.25) calc(var(--rs-card-bezel-x) * 0.1)
+    calc(var(--rs-card-bezel) * -0.25) calc(var(--rs-card-bezel) * -0.4);
+  border-radius: var(--vs-border-radius);
+}
+
+.c-irregularity__icon svg {
+  display: block;
+  width: 70%;
+  height: 70%;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+.c-irregularity__icon svg [fill] {
+  fill: #fff;
+}
+
+.c-irregularity__message {
+  flex: 1 1 auto;
+  font-size: 0.8em;
+  line-height: 1.2;
+}
+
+.c-irregularity__close {
+  position: relative;
+  background: #fff;
+  border-radius: 50%;
+  flex: 0 0 1.5rem;
+  height: 1.5rem;
+  cursor: pointer;
+}
+
+.c-irregularity__close svg {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  width: 100%;
+  transform: translate(-50%, -50%) scale(0.8);
+  transition: 200ms transform cubic-bezier(0.4, 0, 0.2, 1);
+}
+
+.c-irregularity__close:hover svg {
+  transform: translate(-50%, -50%) scale(0.95);
+}
+
+/**
+ *  toast component
+ */
+
+.c-toasts {
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  z-index: var(--vz-toasts);
+}
+
+.c-toast + .c-toast {
+  margin-top: calc(var(--vs-stack) * 0.5);
+}
+
+.c-toast:last-child {
+  margin-bottom: calc(var(--vs-stack) * 0.5);
+}
+
+.c-toast {
+  position: relative;
+  background: var(--rc-background);
+  animation: appearFromBottom 200ms cubic-bezier(0.4, 0, 0.2, 1);
+}
+
+.c-toast--closing {
+  z-index: 1;
+  animation: disappearToBottom 200ms cubic-bezier(0.7, 0.3, 0, 1);
+}
+
+@keyframes appearFromBottom {
+  from {
+    transform: translateY(100%);
```

Comment:
is this 100% of the element height or of the window height? If it's the former, then this assumes the element is at the bottom of the window, right?

Reply:
translate percents always are in relation to the targeted element, so the height of the element. Yes it assumes the element is at the bottom, thats why I called it `appearFromBottom`.